parser/Builder.cc

```diff
@@ -1752,9 +1755,13 @@ class Builder::Impl {
             } else if (auto *blockarg = parser::cast_node<Blockarg>(this_arg.get())) {
                 checkDuplicateArg(blockarg->name, blockarg->loc, map);
             } else if (auto *kwarg = parser::cast_node<Kwarg>(this_arg.get())) {
-                checkDuplicateArg(kwarg->name, kwarg->loc, map);
+                if (checkDuplicateArg(kwarg->name, kwarg->loc, map)) {
+                    kwarg->name = gs_.freshNameUnique(core::UniqueNameKind::MangledKeywordArg, kwarg->name, pos);
+                }
```

Comment:
What do you think about assigning the `checkDuplicateArg` result into a variable so that we can name it?

Right now, `if (checkDuplicateArg)` is kind of ambiguous ("I checked it, and it checks out, so return true" vs "I checked it and it produced an error, so return true").
```suggestion
                auto foundDuplicate = checkDuplicateArg(kwarg->name, kwarg->loc, map);
                if (foundDuplicate) {
                    kwarg->name = gs_.freshNameUnique(core::UniqueNameKind::MangledKeywordArg, kwarg->name, pos);
                }
```

Reply:
Yeah, that's reasonable. Though, what do you think about renaming it to `hasDuplicateArg` instead?